src/components/input/Select.tsx

```diff
@@ -381,10 +381,8 @@ function SelectMain<T>({
     [onChange, closeListbox],
   );
 
-  // When clicking away, focusing away or pressing `Esc`, close the listbox
-  useClickAway(wrapperRef, closeListbox);
+  // Close the listbox when focusing away
   useFocusAway(wrapperRef, closeListbox);
```

Comment:
I guess it would be reasonable, but for now I decided to only mimic what the native popover does by default.

If we see we always end up adding this affordance, in other use cases, we can revisit this decision and add `useFocusAway` to the `Popover`.